plugin-server/src/worker/plugins/run.ts

```diff
@@ -353,15 +353,32 @@ export async function runPluginTask(
 
 async function getPluginMethodsForTeam<M extends keyof VMMethods>(
     hub: Hub,
-    teamId: number,
+    event: ProcessedPluginEvent,
     method: M
 ): Promise<[PluginConfig, VMMethods[M]][]> {
-    const pluginConfigs = hub.pluginConfigsPerTeam.get(teamId) || []
+    let pluginConfigs = hub.pluginConfigsPerTeam.get(event.team_id) || []
     if (pluginConfigs.length === 0) {
         return []
     }
+
+    // Filter based on actionmatching the event against the plugin's configured actions
+    pluginConfigs = (
+        await Promise.all(
+            pluginConfigs.map(async (pluginConfig) => {
+                // TODO: Add matchActions option
+                if (pluginConfig.matchActions.length === 0) {
+                    const matchedActions = await hub.actionMatcher.match(event)
+                    return pluginConfig.matchActions.some((actionId) => matchedActions.find((x) => x.id === actionId))
+                }
+
+                return pluginConfig
+            })
+        )
+    ).filter(Boolean) as PluginConfig[]
```

Comment:
I'd only add this capability to `composeWebhook` plugins and I'd suggest we do this within `runSingleTeamPluginComposeWebhook` <- would make fixing tests a bunch easier probably too

Reply:
Ah shame. Feels like a core functionality but fair enough.

Reply:
there are only two supported kinds of plugins: 
- processEvent  - doesn't make sense to have any filtering here by actions as the actions might not match at that point the same way (the event doesn't have geoIP info yet for example before running geoIP).
- composeWebhook - where we'll be adding this.

We still have some onEvent apps too that we haven't migrated, but there won't be new ones.